ui/editor/src/chessground.js

```diff
@@ -60,16 +63,31 @@ function onMouseEvent(ctrl) {
             piece.color === existingPiece.color &&
             piece.role === existingPiece.role
         ) {
-          pieces[key] = false;
-          ctrl.chessground.setPieces(pieces);
-        } else if (e.type === 'mousedown' || e.type === 'touchstart' || key !== lastKey) {
+          deleteOrHidePiece(ctrl, key, e);
+
+          placeDelete = true;
+
+          if (e.type === 'mousedown') {
+            document.addEventListener('mouseup', function() {
+              placeDelete = false;
+            }, {once: true});
+          } else {
+            document.addEventListener('touchend', function() {
+              placeDelete = false;
+            }, {once: true});
+          }
```

Comment:
code duplication. `var endEvents: {mousedown:'mouseup',touchstart:'touchend'}` would be a way to fix it